### PR TITLE
[stdlib] Fix SIMD uint cast

### PR DESCRIPTION
**Pull request by martinvuyk**
Fix issue #3065 by adding some manual bit padding when converting to uint and forcing the bits to be interpreted as the target type, and slicing when `target.bitwidth() < type.bitwidth()`

Added a bitwidth constraint blocked by issue #933 since the conversion when `target.bitwidth() > type.bitwidth()` uses llvm interleave intrinsic and merging this would cause conflicts for code using `SIMD.cast()` at compile time.

The same constraint also blocks slicing that for some reason doesn't work at compile time when bitcasting the result down to the new type width `failed to fold operation pop.bitcast`

### CODE REVIEW

**Comment by soraros** [comment]
Does this work when `type` is floating point etc.?

**Comment by martinvuyk** [comment]
> Does this work when `type` is floating point etc.?

It doesn't work at all lol,  but no the main issue is `uint -> int` and `int -> uint` casting

**Comment by soraros** [comment]
Shouldn't you also add `type.is_integral()` etc. then?

File: stdlib/src/builtin/simd.mojo [hunk 1]
```diff
@@ -1483,6 +1483,38 @@ struct SIMD[type: DType, size: Int](
...
+                alias zero_x4 = zero_x2.join(zero_x2)
+
+                @parameter
+                if target.bitwidth() // type.bitwidth() == 2:
+                    return bitcast[target, size](zero_x1.interleave(self))
+                elif target.bitwidth() // type.bitwidth() == 4:
+                    return bitcast[target, size](
+                        zero_x2.interleave(zero_x1.interleave(self))
+                    )
+                elif target.bitwidth() // type.bitwidth() == 8:
```
Comment (gryznar) [inline stdlib/src/builtin/simd.mojo:1503]: In every of theses 3 branches, you are calculating `target.bitwidth() // type.bitwidth()`. Separate variable?

**Comment by laszlokindrat** [comment]
Thanks for the patch! I'm currently working on several conversion related issues with SIMD, so I would like to wait with this patch. Is this blocking some other work you're doing ATM?

**Comment by martinvuyk** [comment]
> Is this blocking some other work you're doing ATM?

Nope, just wanted to add some ideas in case it inspired someone who actually understands MLIR :D

**Comment by laszlokindrat** [comment]
Since #3065 has been fixed, this should not be needed anymore. Please open a new ticket if you run into any other problems!

**Comment by martinvuyk** [comment]
this is a very pretty solution to look at :)
```mojo
return self.cast[_uint_type_of_width[int_width]()]().__int__()
```

**Pull request closed without merging**